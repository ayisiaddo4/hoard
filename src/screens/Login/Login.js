import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StatusBar, StyleSheet, Text, Image, View } from 'react-native';
import Config from 'react-native-config';
import T from 'components/Typography';
import LoadingSpinner from 'components/LoadingSpinner';

import { Layout, Body, Header, Footer } from 'components/Layout';
import _ from 'lodash';

import Button from 'components/Button';
import Input from 'components/Input';
import NavigatorService from 'lib/navigator';

const LANG_SIGN_UP_TEXT = 'LOG IN';

export default class Login extends Component {
  static propTypes = {
    hasMnemonic: PropTypes.bool.isRequired,
    navigation: PropTypes.any,
    loginRequest: PropTypes.func.isRequired,
    appReady: PropTypes.bool.isRequired,
    login: PropTypes.shape({
      errors: PropTypes.arrayOf(PropTypes.object),
    }),
  };

  state = {
    loading: false,
    loggedIn: false,
    error: false,
    errorMessage: '',
    username_or_email: null,
    password: null,
  };

  static getDerivedStateFromProps(props, state) {
    if (state.loading && !props.login.requesting && !props.login.successful) {
      return {
        loading: false,
        error: true,
        errorMessage: props.login.errors && props.login.errors[0] && props.login.errors[0].message
      };
    }
    return null;
  }

  handleSignupButton = () => {
    NavigatorService.navigate('Signup');
  };

  handleBypassButton = () => {
    if (this.props.hasMnemonic) {
      NavigatorService.resetReplace('Login', 'Menu');
    } else {
      NavigatorService.navigate('Mnemonic');
    }
  };

  handleFormSubmit = () => {
    this.setState({errorMessage: ''}, () => {
      if (
        this.state.username_or_email &&
        this.state.username_or_email.length > 1 &&
        this.state.password &&
        this.state.password.length > 1
      ) {
        this.setState({ loading: true }, () =>
          this.props.loginRequest({
            username_or_email: this.state.username_or_email,
            password: this.state.password,
          })
        );
      }
    });
  };

  updateFormField = fieldName => text => {
    this.setState({ [fieldName]: text });
  };

  safeFocus = element => _.invoke(element, 'inputRef.focus');

  render() {
    if (!this.props.appReady) {
      return <LoadingSpinner />;
    }

    return (
      <Layout preload={false} keyboard>
        <Body scrollable style={styles.body}>
          <Header style={{ alignItems: 'center' }}>
            <Image
              style={styles.logo}
              source={require('assets/hoard_circle_logo.png')} // eslint-disable-line no-undef
            />
            <Text style={styles.title}>Log In</Text>
            {__DEV__ && (
              <View>
                <T.Small
                  style={styles.network}
                >{`Using: ${Config.ETHNET.toUpperCase()}`}</T.Small>
              </View>
            )}
          </Header>
          <Body>
            {this.state.errorMessage &&
              <View style={styles.errorMessageContainer}>
                <T.Light style={styles.errorMessage}>{this.state.errorMessage}</T.Light>
              </View>
            }
            <Input
              placeholder="Username or Email"
              autoCapitalize="none"
              autoCorrect={false}
              returnKeyType="next"
              keyboardType="email-address"
              editable={!this.state.loading}
              onSubmitEditing={() => this.safeFocus(this.loginPasswordInput)}
              onChangeText={this.updateFormField('username_or_email')}
              value={this.state.username_or_email || ''}
              type="underline"
            />
            <Input
              ref={el => (this.loginPasswordInput = el)}
              placeholder="Password"
              autoCapitalize="none"
              autoCorrect={false}
              returnKeyType="go"
              keyboardType="default"
              editable={!this.state.loading}
              secureTextEntry
              onSubmitEditing={this.handleFormSubmit}
              onChangeText={this.updateFormField('password')}
              value={this.state.password || ''}
              type="underline"
            />
          </Body>
          <Footer>
            <Button type="text" onPress={this.handleSignupButton}>
              New to Hoard? Sign Up!
            </Button>
            <Button
              type="base"
              disabled={!this.state.password && !this.state.username_or_email}
              loading={this.state.loading}
              onPress={this.handleFormSubmit}
              style={styles.buttonContainerAlt}
            >
              {LANG_SIGN_UP_TEXT}
            </Button>
            <Button type="text" onPress={this.handleBypassButton}>
              No thanks, I just want to use the wallet
            </Button>
          </Footer>
        </Body>
      </Layout>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  body: {
    paddingHorizontal: 20,
  },
  containerGradient: {
    borderRadius: 0,
  },

  logoContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    alignContent: 'center',
  },
  logo: {
    width: 150,
    height: 150,
    resizeMode: 'contain',
    marginTop: 10,
    marginRight: 10,
  },
  errorMessageContainer: {
    backgroundColor: '#ff6161',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    borderRadius: 5,
    margin: 20,
  },
  errorMessage: {
    color: '#fff',
    textAlign: 'center',
    fontWeight: 'bold'
  },
  title: {
    color: '#fff',
    marginTop: 10,
    fontSize: 40,
    fontWeight: '100',
    textAlign: 'center',
  },
  network: {
    color: '#fff',
    textAlign: 'center',
  },
  imageView: {
    flex: 1,
    flexShrink: 1,
    paddingTop: 40,
  },
  image: {
    width: null,
    height: null,
    resizeMode: 'cover',
  },
  buttonContainerAlt: {
    marginVertical: 20,
  },
});
