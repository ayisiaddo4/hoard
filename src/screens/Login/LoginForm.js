/**
 *
 * Login Form
 *
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import Error from 'components/notifications/Error';
import Button from 'components/Button';
import Input from 'components/Input';

const LANG_SIGN_UP_TEXT = 'LOG IN';

class LoginForm extends Component {
  static propTypes = {
    navigation: PropTypes.any,
    loginRequest: PropTypes.func.isRequired,
    errors: PropTypes.arrayOf(
      PropTypes.shape({
        code: PropTypes.number,
        message: PropTypes.string,
      })
    ),
  };

  state = {
    loading: false,
    loggedIn: false,
    error: false,
    email_address: null,
    password: null,
  };

  handleSignupButton = () => {
    this.props.navigation.navigate('Signup');
  };

  handleBypassButton = () => {
    if (this.props.hasMnemonic) {
      this.props.navigation.navigate('Menu');
    } else {
      this.props.navigation.navigate('Mnemonic');
    }
  };

  handleFormSubmit = () => {
    if (
      this.state.email_address &&
      this.state.email_address.length > 1 &&
      this.state.password &&
      this.state.password.length > 1
    ) {
      this.props.loginRequest({
        email_address: this.state.email_address,
        password: this.state.password,
      });
    }
  };

  updateFormField = fieldName => text => {
    this.setState({ [fieldName]: text });
  };

  render() {
    const { errors } = this.props;
    return (
      <View style={styles.container}>
        {errors.map(error => {
          return (
            <Error
              key={`error-${error.code}`}
              title="Oops!"
              message={error.message}
            />
          );
        })}
        <Input
          label="Username or Email"
          autoCapitalize="none"
          autoCorrect={false}
          returnKeyType="next"
          keyboardType="email-address"
          onSubmitEditing={() => this.loginPasswordInput.focus()}
          onChangeText={this.updateFormField('email_address')}
          value={this.state.email_address || ''}
          type="underline"
        />
        <View style={{ marginTop: 20 }}>
          <Input
            ref={el => (this.loginPasswordInput = el)}
            label="Password"
            autoCapitalize="none"
            autoCorrect={false}
            returnKeyType="go"
            keyboardType="default"
            secureTextEntry
            onChangeText={this.updateFormField('password')}
            value={this.state.password || ''}
            type="underline"
          />
        </View>
        <Button type="text" onPress={this.handleSignupButton}>
          <Text style={styles.buttonTextBypass}>
            <Text style={styles.buttonTextAlt}>New to Hoard? Sign Up!</Text>
          </Text>
        </Button>

        <View style={{ marginVertical: 20 }}>
          <Button type="primary" onPress={this.handleFormSubmit}>
            {LANG_SIGN_UP_TEXT}
          </Button>
        </View>
        <Button type="text" onPress={this.handleBypassButton}>
          <Text style={styles.buttonTextBypass}>
            No thanks, I just want to use the wallet
          </Text>
        </Button>
      </View>
    );
  }
}

export default LoginForm;

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },

  buttonContainerAlt: {
    backgroundColor: 'transparent',
  },
  buttonTextAlt: {
    textAlign: 'center',
    color: '#fff',
    fontWeight: '700',
  },
  buttonContainerBypass: {
    backgroundColor: 'transparent',
  },
  buttonTextBypass: {
    marginTop: 30,
    textAlign: 'center',
    color: '#fff',
    fontWeight: '500',
  },
});
