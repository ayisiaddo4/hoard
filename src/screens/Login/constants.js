export const LOGIN_REQUESTING = 'login/LOGIN_REQUESTING';
export const LOGIN_SUCCESS = 'login/LOGIN_SUCCESS';
export const LOGIN_ERROR = 'login/LOGIN_ERROR';
