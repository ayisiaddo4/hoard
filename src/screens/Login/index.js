/**
|--------------------------------------------------
| Login Component - connected to redux, and passes
| state & actions to ./Login component.
|--------------------------------------------------
*/
import { connect } from "react-redux";
import loginRequest from "./actions";

import Login from "./Login";

function mapStateToProps(state) {
  return { login: state.login };
}

export default connect(mapStateToProps, { loginRequest })(Login);
