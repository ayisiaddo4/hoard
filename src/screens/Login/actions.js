import { LOGIN_REQUESTING } from './constants';

const loginRequest = function loginRequest({ username_or_email, password }) {
  return { type: LOGIN_REQUESTING, username_or_email, password };
};

export default loginRequest;
