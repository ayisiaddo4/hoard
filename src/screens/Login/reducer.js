import { LOGIN_REQUESTING, LOGIN_SUCCESS, LOGIN_ERROR } from "./constants";

const initialState = {
  requesting: false,
  successful: false,
  messages: [],
  error: null
};

export default function loginReducer(state = initialState, action) {
  switch (action.type) {
    case LOGIN_REQUESTING:
      return { requesting: true, successful: false, messages: [], error: null };

    case LOGIN_SUCCESS:
      return {
        ...state,
        requesting: false,
        successful: true,
        messages: [],
        error: null
      };

    // Append any errors to the current state, if any.
    case LOGIN_ERROR:
      return {
        ...state,
        requesting: false,
        successful: false,
        messages: [],
        error: action.error
      };

    default:
      return state;
  }
}
