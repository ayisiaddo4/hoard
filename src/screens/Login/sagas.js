/**
|--------------------------------------------------
| Login Saga
|--------------------------------------------------
* 
*  This handles all side-effects for logging a user
*  in. Any token storage, anylitics tracking, or app
*  redirection can be triggered here.
*  
*/

import { take, fork, cancel, call, put, cancelled } from "redux-saga/effects";
import { LOGIN_REQUESTING, LOGIN_SUCCESS, LOGIN_ERROR } from "./constants";
import api from "lib/api";

// In order to trigger navigation outside of
// the parent component, we'll use this service.
import NavigatorService from "../../navigator";

// Modify User state
import { setUser, unsetUser } from "screens/User/actions";
import { USER_UNSET } from "screens/User/constants";

// TODO: abstract these into dev/prod files
const loginUrl = `https://smaugdev.hoardinvest.com/login/`;

// Handle logging in via the API
async function loginApi(email_address, password) {
  try {
    const res = await api.post(loginUrl, { email_address, password });
    return { user_uid: res.user_uid };
  } catch (error) {
    throw error;
  }
}

// Handle logging in via the API
// TODO: abstract these into dev/prod files
const logoutUrl = `https://smaugdev.hoardinvest.com/logout/`;
async function logoutApi() {
  try {
    return api.post(logoutUrl);
  } catch (error) {
    throw error;
  }
}

function* logout() {
  // dispatches the USER_UNSET action
  yield call(logoutApi);
  yield put(unsetUser());
  // TODO: remove user session data from Async Storage
  NavigatorService.navigate("Login");
}

function* loginFlow(email_address, password) {
  let currentUser;

  try {
    // try to call to our loginApi() function.  Redux Saga
    // will pause here until we either are successful or
    // receive an error
    currentUser = yield call(loginApi, email_address, password);

    // tell Redux to set currentUser, this is non blocking so...
    yield put(setUser(currentUser));

    // .. we can also tell redux that our login was successful
    yield put({ type: LOGIN_SUCCESS });

    // TODO: store the current user & session data from currentUser in Async Storage

    // redirect the user to the logged in dashboard
    NavigatorService.navigateDeep([
      { routeName: "Menu" },
      { routeName: "Dashboard" }
    ]);
  } catch (error) {
    // TODO: return nice-looking errors to the user
    // yield put({ type: LOGIN_ERROR, error });
  } finally {
    // No matter what, if our `forked` `task` was cancelled
    // we will then just redirect them to login
    if (yield cancelled()) {
      // redirect to the logged out screen
      NavigatorService.navigate("Login");
    }
  }

  // return the currentUser for health and wealth
  return currentUser;
}

// Our watcher (saga).  It will watch for all login-related actions.
function* loginWatcher() {
  // Generators halt execution until their next step is ready/occurring
  while (true) {
    // ... and in this first it sees a yield statement which
    // pauses the loop.  It will sit here and WAIT for this action.
    //
    // yield take(ACTION) just says, when our generator sees the ACTION
    // it will pull from that ACTION's payload that we send up, its
    // email and password.  ONLY when this happens will the loop move
    // forward...
    const { email_address, password } = yield take(LOGIN_REQUESTING);
    // ... and pass the email and password to our loginFlow() function
    // the fork() method spins up another "process" that will deal with
    // handling the loginFlow's execution in the background!
    // Think, "fork another process".
    //
    // It also passes back to us, a reference to this forked task
    // which is stored in our const task here.  We can use this to manage
    // the task.
    //
    // However, fork() does not block our loop.  It's in the background
    // therefore as soon as our loop executes this it mores forward...
    const task = yield fork(loginFlow, email_address, password);

    // ... and begins looking for either USER_UNSET or LOGIN_ERROR!
    // That's right, it gets to here and stops and begins watching
    // for these tasks only.  Why would it watch for login any more?
    // During the life cycle of this generator, the user will login once
    // and all we need to watch for is either logging out, or a login
    // error.  The moment it does grab either of these though it will
    // once again move forward...
    const action = yield take([USER_UNSET, LOGIN_ERROR]);

    // ... if, for whatever reason, we decide to logout during this
    // cancel the current action.  i.e. the user is being logged
    // in, they get impatient and start hammering the logout button.
    // this would result in the above statement seeing the USER_UNSET
    // action, and down here, knowing that we should cancel the
    // forked `task` that was trying to log them in.  It will do so
    // and move forward...
    if (action.type === USER_UNSET) yield cancel(task);

    // ... finally we'll just log them out.  This will unset the client
    // access token ... -> follow this back up to the top of the while loop
    yield call(logout);
  }
}

export default loginWatcher;
