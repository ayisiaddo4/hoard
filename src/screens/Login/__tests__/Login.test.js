import React from 'react';
import Login from "../Login";
import renderer from "react-test-renderer";


function snapshotForm(props) {
  const tree = renderer.create(<Login {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
}

describe("Login view", () => {
  it("should render correctly", () => {
    let props = {
        navigator: {},
        login: {errors: []},
        loginRequest: ()=>{}
    };
    snapshotForm(props);
  });
});
