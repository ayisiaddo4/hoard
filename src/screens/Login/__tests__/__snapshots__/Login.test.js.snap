// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login view should render correctly 1`] = `
<View
  imageStyle={
    Object {
      "height": null,
      "resizeMode": "cover",
      "width": null,
    }
  }
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={false}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingTop": 40,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            Object {
              "height": null,
              "resizeMode": "cover",
              "width": null,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 40,
              "marginRight": 10,
              "marginTop": 10,
              "resizeMode": "contain",
              "width": 40,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 40,
              "fontWeight": "100",
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          Log In
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignItems": "center",
            "color": "#fff",
            "flex": 1,
            "fontSize": 30,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        Using: ROPSTEN
      </Text>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <View
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="you@email.com"
            placeholderTextColor="rgba(255,255,255,0.75)"
            returnKeyType="next"
            style={
              Object {
                "backgroundColor": "rgba(0,0,20, 0.25)",
                "borderRadius": 8,
                "color": "#fff",
                "height": 40,
                "marginBottom": 15,
                "paddingHorizontal": 10,
              }
            }
          />
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="password"
            placeholderTextColor="rgba(255,255,255,0.75)"
            returnKeyType="go"
            secureTextEntry={true}
            style={
              Object {
                "backgroundColor": "rgba(0,0,20, 0.25)",
                "borderRadius": 8,
                "color": "#fff",
                "height": 40,
                "marginBottom": 15,
                "paddingHorizontal": 10,
              }
            }
          />
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 100,
                "marginBottom": 10,
                "opacity": 1,
                "paddingVertical": 20,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 4,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 12,
                    "letterSpacing": 0,
                    "lineHeight": 18,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#000",
                    "textAlign": "center",
                  },
                ]
              }
            >
              LOG IN
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "opacity": 1,
                "paddingVertical": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "700",
                  "textAlign": "center",
                }
              }
            >
              New to Hoard? Sign Up!
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "marginTop": 50,
                "opacity": 1,
                "paddingVertical": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "500",
                  "textAlign": "center",
                }
              }
            >
              No thanks, I just want to use the wallet
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
